src/store/ViewStore.ts

```diff
@@ -159,10 +160,12 @@ export default class ViewStore {
         this.mainStore.state.setChartIsReady(false);
         const sortedItems = this.sortedItems[idx].layout;
         const stx = this.stx;
-        const granularity = getIntervalInSeconds(sortedItems) as TGranularity;
+        const { interval, periodicity, timeUnit } = sortedItems;
+        const period = timeUnit ? interval : periodicity;
+        const granularity = calculateGranularity(Number(period), (timeUnit as string) || (interval as string));
 
-        this.mainStore.timeperiod.onGranularityChange(granularity);
```

Comment:
we don't need to call this callback since it's not assigned with any function unless we open Chart mode modal to change Chart type of Time interval before applying a template.
`this.mainStore.state.setChartGranularity(granularity);` at L177 serves this purpose well enough.